mlrun/config.py

```diff
@@ -143,6 +143,17 @@
     # when set (True or non empty str) it will force the mock=True in deploy_function(),
     # set to "auto" will use mock of Nuclio if not detected (no nuclio_version)
     "mock_nuclio_deployment": "",
+    # Configurations for `mlrun.track` - tracking runs and experiments from 3rd party vendors like MLFlow
+    # by running them as a MLRun function, capturing their logs, results and artifacts to mlrun.
+    "external_platform_tracking": {
+        # General enabler for the entire tracking mechanism (all tracking services):
+        "enabled": False,
+        # Specific enablement and other configurations supported 3rd party platforms:
+        "mlflow": {
+            # Add the MLFlow tracker to MLRun:
+            "mode": "enabled"
```

Comment:
Is there any other possible mode rather than `enabled` (or anything else for `disabled`)?
I'm thinking you can do instead `"enabled": True` which is easier to understand and doesn't make you wonder whether there are other modes... It is also consistent with the global enablement for the tracker feature.

Reply:
The idea here is to allow adding modes in the future. If it will be a boolean and someday we want to introduce a new mode like partial tracking or whatever, we'll need to deprecate it.

Reply:
Logically, `enabled` is a yes-no question and it's unrelated to operational mode. If we add modes in the future then we can add a `mode` parameter that identifies the mode, and enable you to still enable/disable a tracker regardless of its mode.

Reply:
you want me to change that? @theSaarco

Reply:
@guy1992l - my opinion is written above, I think this should be changed.

Reply:
```python
# by running them as a MLRun function, capturing their logs, results and artifacts to mlrun.
    "external_platform_tracking": {
        # General enabler for the entire tracking mechanism (all tracking services):
        "enabled": False,
        # Specific enablement and other configurations supported 3rd party platforms:
        "mlflow": {
            # Enable of disable MLFlow tracking:
            "enabled": True,
            # MLFlow tracking mode:
            "mode": "full",
        },
    },
```

how's this?

Reply:
My suggestion was the opposite - we don't have any meaningful `mode` here, and the only thing we might want to do is indicate that this platform is enabled/disabled. So I was suggesting to keep **only** the `enabled` param for now. If at some point in the future we'll have a mode, we can add it. This will allow setting both the mode and deciding whether we want to enable or disable this feature.

Reply:
gotcha